### ISSUE TITLE
[feature request] proxy settings

### ISSUE THREAD

**Issue by Riajyuu** [digimezzo/dopamine-windows#381]
Allow users to define a proxy through which Dopamine connect Internet.

**Comment by digimezzo** [comment]
Doesn't Dopamine use the default proxy? I thought that worked.

**Comment by Riajyuu** [comment]
Yes, indeed. But there are times we don't change default proxy but use a local hosts address (specific example: shadowsock listens 127.0.0.1 at port 1080).
In those cases, Dopamine never use those proxy

**Comment by Magentaize** [comment]
Maybe @EMLVIRUS wants a build-in socks5 proxy support.

**Comment by digimezzo** [comment]
I'm not sure what you want? :) Anyway, I guess that would require adding some fields for hostname and port, an than extend the httpclient to allow usage of a proxy.

**Comment by Riajyuu** [comment]
>  adding some fields for hostname and port, an than extend the httpclient to allow usage of a proxy.

Yes, that's exactly what I'm requesting

**Comment by Magentaize** [comment]
I recommend two applications to you, they are Proxifier and SocksCap64. If you want some apps which don't support socks proxy can use your proxies, they are the best.
